app/Models/Article.php

```diff
@@ -108,4 +148,25 @@ public function getSlugOptions(): SlugOptions
             ->saveSlugsTo('slug')
             ->doNotGenerateSlugsOnUpdate();
     }
+
+    public static function updateViewCounts(): void
+    {
+        $now = now();
+        $pastWeek = now()->subDays(7);
+
+        Article::query()
+            ->update([
+                'views_count_7days' => DB::raw(
+                    <<<SQL
+(
+    SELECT COUNT(*) FROM views AS v
```

Comment:
what about using https://github.com/cyrildewit/eloquent-viewable#get-views-count-of-a-specific-period for this as opposed to a raw query

Reply:
if we take the given approach we will have to pull all models (articles) and then do the update:

1. step:
```
// Example: get views count from 2017 upto 2018
views($post)
    ->period(Period::create('2017', '2018'))
    ->count();
```
2.nd step:
```
$posts = Post::all();

foreach($posts as $post) {
    $post->unique_views_count = views($post)->unique()->count();
}
```

While the crafted query will do it on the fly. I can still implement this solution but it will be more resource intensive than the current one.